py/kubeflow/kfctl/testing/pytests/endpoint_ready_test.py

```diff
@@ -41,6 +41,19 @@ def test_endpoint_is_ready(record_xml_attribute, project, app_path, app_name, us
     if not gcp_util.iap_is_ready(url):
       raise Exception("IAP endpoint is not ready")
 
+# Todo(jlewi): Should this only run in postsubmit?
+@pytest.mark.skipif(os.getenv("JOB_TYPE") == "presubmit",
+                    reason="test endpoint doesn't run in presubmits")
+def test_central_dash_is_running(record_xml_attribute, app_path, kf_host, client_id, service_acct_email, service_acct_key_file):
+  """Test that Kubeflow Central Dash was successfully deployed."""
+  util.set_pytest_junit(record_xml_attribute, "test_central_dash_is_running")
+  os.environ["KF_HOST"] = kf_host       # The url to your cluster, usually: https://<cluster>.endpoints.<project>.cloud.goog/
+  os.environ["CLIENT_ID"] = client_id   # This is the CLIENT_ID token used to create the cluster (used in http://deploy.kubeflow.cloud/)
+  os.environ["SERVICE_ACCOUNT_EMAIL"] = service_acct_email  # The email address for the service account you created <name>@<project>.iam.gserviceaccount.com
+  os.environ["SERVICE_ACCOUNT_KEY"] = service_acct_key_file # Should be a JSON file downloaded from cloud console service accounts
+
+  subprocess.check_call('cd "{}" && npm run test-e2e'.format(app_path), shell=True)
```

Comment:
Per our discussion wrapping this in pytest is fine. I wonder if this is an unnecessary layer of direction and if its worth replacing right now or in a follow on PR.

PyTest is basically buying us the following functionality

1. It produces a junit XML file with the results which we need to surface results in prow
1. It is easy to mark the test as skippable?

It looks like the test is using the jasmine framework; does that framework provide this functionality?

I was originally thinking that by adding this test to endpoint_ready_test.py we could avoid the complexity of adding a step to the argo workflow. I suspect though we may not be able to avoid that. 

Since the test depends on IAP, we probably need to run this after the endpoint is ready test to make sure the IAP endpoint is available. Its possible there is a way to do this using pytest but otherwise we may need to modify the Argo graph.


/cc @kimwnasptd

Reply:
Seems like it can: https://www.npmjs.com/package/jasmine-xml-reporter